src/errors.rs

```diff
@@ -0,0 +1,33 @@
+use std::str::Utf8Error;
+
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, ArklibError>;
+
+#[derive(Error, Debug)]
+pub enum ArklibError {
+    #[error("IO error")]
+    Io(#[from] std::io::Error),
+    #[error("Invalid path for ressource")]
```

Comment:
Resource with one `s` :wink:  I already fixed that once, so now telling in comments..